src/dbus_api/pool/shared.rs

```diff
@@ -197,3 +199,27 @@ pub fn pool_name_prop(name: &Name) -> String {
 pub fn pool_enc_prop(pool: &dyn Pool) -> bool {
     pool.is_encrypted()
 }
+
+/// Get the property changed object that should be constructed if a pool name
+/// is changed.
+pub fn get_name_change_properties(pool_name: &Name) -> Vec<PropertiesPropertiesChanged> {
+    let mut properties_changed: Vec<PropertiesPropertiesChanged> = vec![];
+
+    let mut r0_properties = PropertiesPropertiesChanged::default();
+    r0_properties.changed_properties.insert(
+        consts::POOL_NAME_PROP.into(),
+        Variant(Box::new(pool_name_prop(pool_name))),
+    );
+    r0_properties.interface_name = consts::POOL_INTERFACE_NAME.into();
+    properties_changed.push(r0_properties);
+
+    let mut r1_properties = PropertiesPropertiesChanged::default();
+    r1_properties.changed_properties.insert(
+        consts::POOL_NAME_PROP.into(),
+        Variant(Box::new(pool_name_prop(pool_name))),
+    );
+    r1_properties.interface_name = consts::POOL_INTERFACE_NAME.into();
```

Comment:
Should this be `POOL_INTERFACE_NAME_2_1`?

Reply:
Fixed. But there's also a further thing I'ld like to try, to move more processing into the ActionQueue implementation.